homeassistant/components/wilight/parent_device.py

```diff
@@ -0,0 +1,110 @@
+"""The WiLight Device integration."""
+import asyncio
+import logging
+
+import pywilight
+import requests
+
+from homeassistant.const import CONF_HOST, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WiLightParent:
+    """Manages a single WiLight Parent Device."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the system."""
+        self._host = config_entry.data[CONF_HOST]
+        self._hass = hass
+        self._api = None
+
+    @property
+    def host(self):
+        """Return the host of this parent."""
+        return self._host
+
+    @property
+    def api(self):
+        """Return the api of this parent."""
+        return self._api
+
+    async def async_setup(self):
+        """Set up a WiLight Parent Device based on host parameter."""
+        host = self._host
+        hass = self._hass
+
+        api_device = await hass.async_add_executor_job(create_api_device, host, 45995)
```

Comment:
Why is this done in the executor while the rest of the api calls are awaited? Does the library support asyncio?

Reply:
Honestly, this was the best way I did to integrate after several tests.
Yes, the library supports asyncio.

Reply:
Do we need to make a request to the URL or can we create the device from the info we get from the SSDP discovery in the config flow? We have at least host, serial number and model.

Reply:
I was wrong when I replied that the library supports asyncio.
In fact, the part that deals with discovery, until creating the device, does not.
Who supports asyncio is the communication part (API) after creating the device
Hence the need for async_add_executor_job.
Today I can even create a device from host, serial number and model, but I would also need to get model_number from ssdp.
It turns out that I plan to get more device information in the future to import the names of the child devices from my application. Homeassinstant SSDP does not support this.
Therefore, I intend to keep async_add_executor_job if that is not a hindrance to approve the integration.

Reply:
I recommend using aiohttp for the requests instead of requests. Then the library would be totally async. It's not a requirement.

Reply:
How this will imply major changes and tests in the library. I will try, but I would like it to go to a new PR when I add new components.
Can we approve the current one as it is?

Reply:
Yes, it's ok as is.